storage/reads/mergegroupresultset_test.go

```diff
@@ -225,6 +225,24 @@ group:
 			},
 			exp: exp,
 		},
+		{
+			name: "does merge keys",
```

Comment:
I'm having trouble understanding this test case. What exactly is different about this one vs the other two test cases?

Reply:
@jlapacik, I have added a long description to this PR to describe the new test case.